sympy/solvers/tests/test_diophantine.py

```diff
@@ -583,6 +582,58 @@ def check_solutions(eq):
 
     return okay
 
+
+def check_linear(eq, param=symbols("t", Integer=True)):
```

Comment:
The extra time probably comes from the extra tests I added, such as those for univariate linear diophantine equations, because _diop_solve() now handles them (as you suggested it should).

I only see a 7% increase in execution time, comparing apples to apples:

```
    iters = 1000

    start = time.clock()
    for i in range(iters):
        assert diop_solve(x) == (0,)
        assert diop_solve(1*x) == (0,)
        assert diop_solve(3*x) == (0,)
        assert diop_solve(x + 1) == (-1,)
        assert diop_solve(2*x + 1) == (None,)
        assert diop_solve(2*x + 4) == (-2,)
        assert diop_solve(y + x) == (t_0, -t_0)
        assert diop_solve(y + x + 0) == (t_0, -t_0)
        assert diop_solve(y + x - 0) == (t_0, -t_0)
        assert diop_solve(0*x - y - 5) == (-5,)
        assert diop_solve(3*y + 2*x - 5) == (3*t_0 - 5, -2*t_0 + 5)
        assert diop_solve(2*x - 3*y - 5) == (-3*t_0 - 5, -2*t_0 - 5)
        assert diop_solve(-2*x - 3*y - 5) == (-3*t_0 + 5, 2*t_0 - 5)
        assert diop_solve(7*x + 5*y) == (5*t_0, -7*t_0)
        assert diop_solve(2*x + 4*y) == (2*t_0, -t_0)
        assert diop_solve(4*x + 6*y - 4) == (3*t_0 - 2, -2*t_0 + 2)
        assert diop_solve(4*x + 6*y - 3) == (None, None)
        assert diop_solve(0*x + 3*y - 4*z + 5) == (-4*t_0 + 5, -3*t_0 + 5)
        assert diop_solve(4*x + 3*y - 4*z + 5) == (t_0, -4*t_1 + 5, t_0 - 3*t_1 + 5)
        assert diop_solve(4*x + 2*y + 8*z - 5) == (None, None, None)
        assert diop_solve(5*x + 7*y - 2*z - 6) == \
            (t_0, -7*t_0 - 2*t_1 + 6, -22*t_0 - 7*t_1 + 18)
        assert diop_solve(3*x - 6*y + 12*z - 9) == (2*t_0 + 3, t_0 + 2*t_1, t_1)
        assert diop_solve(x + 3*y - 4*z + w - 6) == \
            (t_0, t_0 + t_1, -2*t_0 - 3*t_1 - 4*t_2 - 6, -t_0 - 2*t_1 - 3*t_2 - 6)
        assert diop_solve(2*w + 3*x + 5*y + 7*z + 11*t + 13*X + 17*Y + 19*Z - 23) == \
            (t_0, t_0 + t_1, t_0 + t_1 + t_2, t_0 + t_1 + t_2 + t_3, t_0 + t_1 + t_2 + t_3 + t_4, t_0 + t_1 + t_2 + t_3 + t_4 + t_5, -188*t_0 - 149*t_1 - 98*t_2 - 41*t_3 - 8*t_4 - 2*t_5 + 7*t_6 + 69, 125*t_0 + 99*t_1 + 65*t_2 + 27*t_3 + 5*t_4 + t_5 - 5*t_6 - 46)
        assert diop_solve(2*w + 4*x + 6*y + 8*z + 10*t + 12*X + 14*Y + 16*Z - 1) == \
            (None, None, None, None, None, None, None, None)
    end = time.clock()
    print("first: " + str(end - start))
    
    start = time.clock()
    for i in range(iters):
        assert check_linear(x)
        assert check_linear(1*x)
        assert check_linear(3*x)
        assert check_linear(x + 1)
        assert check_linear(2*x + 1)
        assert check_linear(2*x + 4)
        assert check_linear(y + x)
        assert check_linear(y + x + 0)
        assert check_linear(y + x - 0)
        assert check_linear(0*x - y - 5)
        assert check_linear(3*y + 2*x - 5)
        assert check_linear(2*x - 3*y - 5)
        assert check_linear(-2*x - 3*y - 5)
        assert check_linear(7*x + 5*y)
        assert check_linear(2*x + 4*y)
        assert check_linear(4*x + 6*y - 4)
        assert check_linear(4*x + 6*y - 3)
        assert check_linear(0*x + 3*y - 4*z + 5)
        assert check_linear(4*x + 3*y - 4*z + 5)
        assert check_linear(4*x + 2*y + 8*z - 5)
        assert check_linear(5*x + 7*y - 2*z - 6)
        assert check_linear(3*x - 6*y + 12*z - 9)
        assert check_linear(x + 3*y - 4*z + w - 6)
        assert check_linear(2*w + 3*x + 5*y + 7*z + 11*t + 13*X + 17*Y + 19*Z - 23)
        assert check_linear(2*w + 4*x + 6*y + 8*z + 10*t + 12*X + 14*Y + 16*Z - 1)
    end = time.clock()
    print("second: " + str(end - start))
```

Output:

```
first: 16.363018
second: 17.423285
```